src/go/rpk/pkg/config/config_test.go

```diff
@@ -551,6 +551,51 @@ redpanda:
   target_quota_byte_rate: 1000000
 unrecognized_top_field:
   child: true
+`,
+		},
+		{
+			name: "should merge the new config onto the current one, not just overwrite it",
+			existingConf: `config_file: /etc/redpanda/redpanda.yaml
+redpanda:
+  admin:
+    address: 0.0.0.0
+    port: 9644
+  admin_api_doc_dir: /etc/redpanda/doc
+  auto_create_topics_enabled: true
+  data_directory: /var/lib/redpanda/data
+  default_window_sec: 100
+  kafka_api:
+    address: 0.0.0.0
+    port: 9092
+  node_id: 0
+  rpc_server:
+    address: 0.0.0.0
+    port: 33145
+  target_quota_byte_rate: 1000000
+`,
+			conf: func() *Config {
+				conf := getValidConfig()
+				conf.Redpanda.SeedServers = []SeedServer{}
```

Comment:
Should this test override a value to test which way the merge happens?

Reply:
Done!